idf_build_apps/app.py

```diff
@@ -110,6 +121,11 @@ class App(BaseModel):
     build_status: BuildStatus = BuildStatus.UNKNOWN
     build_comment: t.Optional[str] = None
 
+    # SDKCONFIG_DEFAULT_FILE_OVERRITE
+    override_sdkconfig: t.Optional[dict[str, Any]] = {}
+    override_sdkconfig_file: t.Optional[tuple[str]] = ()
+    override_priority: OverridePriority = OverridePriority.OVERRIDE_SDKCONFIG
```

Comment:
pydantic treat these as instance attributes, if you're going to use `App.override_sdkconfig` later, which is a class instance, you should use `t.ClassVar` instead.

I think we could set two things separately, take `override_sdkconfig` as an example

- set instance attr `override_sdkconfig: t.Optional[dict[str, Any]] = None`
- set session param `--override-sdkconfig` in `BuildJob`

```python
app = App(...)
app.override_sdkconfig = override_sdkconfig is override_sdkconfig is not None else BuildJob().override_sdkconfig
```

Reply:
Yes, we can place the `override_sdkconfig` in both the `App` and `BuildJob`. However, for attributes like `override_sdkconfig_file` and `override_priority`,  we should only assign them to the `BuildJob`.

Reply:
Thank you for your comment about `t.ClassVar`! :+1:  I've used it in `App` for both instance and class attributes, but when we move the attributes to `BuildJob`, I'd like to set it up there as a `t.ClassVar`.